velox/functions/sparksql/aggregates/BloomFilterAggAggregate.cpp

```diff
@@ -102,8 +94,8 @@ class BloomFilterAggAggregate : public exec::Aggregate {
     computeCapacity();
     auto mayHaveNulls = decodedRaw_.mayHaveNulls();
     rows.applyToSelected([&](vector_size_t row) {
-      if (mayHaveNulls) {
-        checkBloomFilterNotNull(decodedRaw_, row);
+      if (mayHaveNulls && decodedRaw_.isNullAt(row)) {
```

Comment:
Could be optimized similarly by separating into different branches. Maybe wrap the common part in a lambda to avoid copy-paste.